### ISSUE TITLE
Bug: When update nested array of object

### ISSUE THREAD

**Issue by zhaoyao91** [jagi/meteor-astronomy#181]
for test reason, I have such a model:

    Items = new Mongo.Collection('items');
    
    Item = Astro.Class({
        name: 'Item',
        collection: Items,
        fields: {
            subItems: {
                type: 'array',
                nested: {
                    name: 'SubItem',
                    fields: {
                        subItems: {
                            type: 'array',
                            nested: {
                                name: 'SubSubItem',
                                fields: {
                                    images: {type: 'array', nested: 'string', default(){return []}},
                                    image: {type: 'string'}
                                }
                            }
                        }
                    }
                }
            }
        }
    });

while I can insert correctly: `Item.insert({subItems:[{subItems:[{images:['1']}]}]})`
![image](https://cloud.githubusercontent.com/assets/3808838/11019450/b3339d5c-8634-11e5-8e25-a3e1bf86795b.png)

I can't update it correctlly: `Item.update('hFM2u6vBdt2r9qbc3', {$set:{subItems:[{subItems:[{images:['2']}]}]
}})`
![image](https://cloud.githubusercontent.com/assets/3808838/11019452/d5b8d360-8634-11e5-9b1e-4b74f78e2ce0.png)

**Comment by jagi** [comment]
You have to have an error somewhere in your code. Create a reproduction. If I use the following code, it works:

```js
Items = new Mongo.Collection(null);

Item = Astro.Class({
    name: 'Item',
    collection: Items,
    fields: {
        subItems: {
            type: 'array',
            nested: {
                name: 'SubItem',
                fields: {
                    subItems: {
                        type: 'array',
                        nested: {
                            name: 'SubSubItem',
                            fields: {
                                images: {type: 'array', nested: 'string', default(){return []}},
                                image: {type: 'string'}
                            }
                        }
                    }
                }
            }
        }
    }
});

Item.insert({subItems:[{subItems:[{images:[1]}]}]});
var item = Item.findOne();
Item.update(item._id, {subItems:[{subItems:[{images:[2]}]}]});
```

**Comment by zhaoyao91** [comment]
sorry, but your code doesn't work, your update modifier is wrong. In meteor, you must use $set or something else, but you can't just replace the doc.

here is the result of your doc. note the update has no effect:
![image](https://cloud.githubusercontent.com/assets/3808838/11020002/fe4d4d8a-864b-11e5-9d50-c322f2c7cb1a.png)

**Comment by jagi** [comment]
Sorry I've pasted wrong code

**Comment by jagi** [comment]
Oh yes even by fixing code, it does not work properly. I haven't noticed that the problem is about how data is stored in most nested field. I will have to investigate it further.

**Comment by jagi** [comment]
Ok it will be fixed in 1.2.1 which is going to be publish today

**Comment by zhaoyao91** [comment]
thanks
